test/unit/gateways/cyber_source_test.rb

```diff
@@ -737,6 +737,25 @@ def test_successful_verify
     assert_success response
   end
 
+  def test_successful_verify_0_amount
```

Comment:
💬  Comment:

Couple of things here:

1) You are mocking the :ssl_request instead of the :ssl_post, that's why you have the request body on the headers variable.
2) you don't need to assert the response when you are checking something in the request.

You want something like this:

```Ruby
  def test_successful_verify_0_amount
    @options[:zero_dollar_auth] = true
    stub_comms(@gateway, :ssl_post) do
      @gateway.verify(@credit_card, @options)
    end.check_request(skip_response: true) do |_endpoint, data|
      assert_match %r(<grandTotalAmount>0.00<\/grandTotalAmount>), data
    end
  end

```

Reply:
Oh, I see, I'll use :ssl_post instead of :ssl_request.